### PR TITLE
Add accept header check for prometheus mime type

### LINKED ISSUE #7936: Prometheus Content-Type negotiation
**Is your feature request related to a problem? Please describe.**
Scraping Prometheus metrics requires the query parameter `?format=prometheus`. In a Kubernetes setup, scraping endpoints tend be discovered using pod annotations (e.g. `prometheus.io/path`), however an annotation for parameters is not very common (`?` is escaped properly by Prometheus, so parameters cannot be provided in the path). 

**Describe the solution you'd like**
Prometheus does Content-Type negotiation (see [here](https://github.com/prometheus/prometheus/issues/1417#issuecomment-187297420)). When Vault sees the schema `schema="prometheus/telemetry"` it could serve Prometheus metrics without having to specify the `format` parameter.

**Describe alternatives you've considered**
None

**Explain any additional use-cases**
None

### PR DESCRIPTION
**Pull request by michelvocks**
Fixes #7936

### CODE REVIEW

File: helper/metricsutil/metricsutil.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ import (
 const (
 	OpenMetricsMIMEType = "application/openmetrics-text"
 
+	PrometheusSchemaMIMEType = "prometheus/telemetry"
```
Comment (briankassouf) [inline helper/metricsutil/metricsutil.go]: This was added, but doesn't seem to be used anywhere

File: helper/metricsutil/metricsutil.go [hunk 2]
```diff
@@ -38,7 +40,14 @@ func FormatFromRequest(req *logical.Request) string {
 	if len(acceptHeaders) > 0 {
 		acceptHeader := acceptHeaders[0]
 		if strings.HasPrefix(acceptHeader, OpenMetricsMIMEType) {
-			return "prometheus"
+			return PrometheusMetricFormat
+		}
+
+		// Look for prometheus accept header
+		for _, header := range acceptHeaders {
+			if strings.Contains(header, PrometheusMetricFormat) {
```
Comment (briankassouf) [inline helper/metricsutil/metricsutil.go]: Guessing that this should be `PrometheusSchemaMIMEType` here? Also can we add a test for this?

File: helper/metricsutil/metricsutil.go (hunk 1 above)
Comment (michelvocks) [reply helper/metricsutil/metricsutil.go]: Good catch! Fixed

File: helper/metricsutil/metricsutil.go (hunk 2 above)
Comment (michelvocks) [reply helper/metricsutil/metricsutil.go]: done

**Pull request merged**